expfmt/decode.go

```diff
@@ -107,7 +108,30 @@ type protoDecoder struct {
 // Decode implements the Decoder interface.
 func (d *protoDecoder) Decode(v *dto.MetricFamily) error {
 	_, err := pbutil.ReadDelimited(d.r, v)
-	return err
+	if err != nil {
+		return err
+	}
+	if !utf8.ValidString(v.GetName()) {
```

Comment:
We have a regexp somewhere for metric names. We should check against the regexp here (which will include checking for valid UTF-8).